pkg/autoscaler/metrics/collector.go

```diff
@@ -385,6 +370,41 @@ func (c *collection) record(stat Stat) {
 	c.rpsPanicBuckets.Record(stat.Time, rps)
 }
 
+// startScraping starts scraping metrics.
+// This is only run from inside a lock, it can't race with a stopScraping call.
+func (c *collection) startScraping() {
+	c.stopCh = make(chan struct{})
+	c.grp.Add(1)
+	go func() {
+		defer c.grp.Done()
+
+		scrapeTicker := c.tickFactory(scrapeTickInterval)
+		defer scrapeTicker.Stop()
```

Comment:
join defers into one call?

Reply:
Doesn't feel like we gain much doing that, but we lose correct behavior if `tickFactory` panic, I think?

Reply:
if it panics we have worse problems 😁